### PR TITLE
testing refresh of d-acts import

### PR DESCRIPTION
**Pull request by wdduncan**
- replace http://purl.obolibrary.org/obo/iao/d-acts/release/2020-10-14/d-acts.owl with http://purl.obolibrary.org/obo/iao/d-acts.owl
- add new d-acts

### CODE REVIEW

**Comment by wdduncan** [comment]
@hoganwr @dillerm @CDowland 

I created this PR by forking the OMRSE repo. Here is a summary of what is going on when using ODK to refresh the d-acts import.

When you execute`make refresh-d-acts` on your machine, the relevant part of the `makefile` that runs is this:
```
$(IMPORTDIR)/d-acts_import.owl: $(MIRRORDIR)/d-acts.owl $(IMPORTDIR)/d-acts_terms_combined.txt
        if [ $(IMP) = true ]; then $(ROBOT) merge -i $< \
                query --update ../sparql/preprocess-module.ru --update ../sparql/inject-subset-declaration.ru --update ../sparql/postprocess-module.ru \
                annotate --ontology-iri $(ONTBASE)/$@ $(ANNOTATE_ONTOLOGY_VERSION) --output $@.tmp.owl && mv $@.tmp.owl $@; fi
```
Unpacking this gave me a headache, but here is my crude summary:
- The goal/dependency `$(MIRRORDIR)/d-acts.owl` will create a `mirror` directory and download `d-acts.owl` into it. The `mirror` is only present on your local machine, and not on the remote repo. This is b/c in the `omrse/.gitignore` file the `src/ontology/mirror` is specified as being ignored.
- The rest of the make command calls `robot` to do a bunch of processing on the `mirror/d-act.owl` file in order to create the `imports/d-act_import.owl` file.

The first time I ran `make refresh-d-acts` nothing changed in regards to the number of definitions for `aggregate of organizations`. I found this odd, but then noticed that command for creating the d-acts mirror was tied to a specific version of the ontology:
```
mirror-d-acts: | $(TMPDIR)
     if [ $(MIR) = true ] && [ $(IMP) = true ]; then $(ROBOT) 
     convert -I http://purl.obolibrary.org/obo/iao/d-acts/release/2020-10-14/d-acts.owl 
    -o $@.tmp.owl && mv $@.tmp.owl $(TMPDIR)/$@.owl; fi
```
Notice the `release/2020-10-14/d-acts.owl` part of the IRI. 

I don't know if locking on to this specific release was by design, but I changed the Makefile to pull the most current version of d-acts. See line `454` in the  Makefile diff.

Now, after running `make refresh-d-acts`, a new `d-acts_import.owl` file was created. Because `d-acts_import.owl` is saved as rdf/xml, the order in which rdf/xml is written is not consistent. This can (and does) create a bunch of spurious diffs (I know ... annoying). But, if go to lines `8583-8651` of the `d-acts_import.owl` diff, you will indeed see that two definitions have been replaced by a single definition. Here is a screenshot.

image

So did this fix the problem? ... **NO!** If you open the `omrse-edit.owl` file you see that `aggregate of organizations` still has two definitions (grrr...).

image

The reason for this is b/c `omrse-edit.owl` a definition `aggregate of organizations` has been created within `omrse-edit.owl`. To see this, select the option to `Show only the active ontology` in Protege:

image

Now you see the **single** definition that has been created within the `omrse-edit.owl` file:

image

However, since `omrse-edit.owl` is importing `imports/d-acts_imports.owl` and `imports/d-acts_imports.owl` has a different definition for `aggregate of organizations`, two definitions will be present when you view `omrse-edit.owl` using the `Show the imports closure of the active ontology` option in Protege (which is the viewing option I most commonly use):

image

So, at this point, it is unclear what is the source of truth for the definition of `aggregate of organizations`. Is it one in the most recent release of `d-acts` or the one `omrse`?

**Comment by mbrochhausen** [comment]
I am not sure whether this is a d-acts issue. If so , please let me know and I fix it asap.

Best,
Mathias

File: src/ontology/Makefile [hunk 1]
```diff
@@ -451,7 +451,7 @@ mirror-pno: | $(TMPDIR)
 .PHONY: mirror-d-acts
 .PRECIOUS: $(MIRRORDIR)/d-acts.owl
 mirror-d-acts: | $(TMPDIR)
-	if [ $(MIR) = true ] && [ $(IMP) = true ]; then $(ROBOT) convert -I http://purl.obolibrary.org/obo/iao/d-acts/release/2020-10-14/d-acts.owl -o $@.tmp.owl && mv $@.tmp.owl $(TMPDIR)/$@.owl; fi
```
Comment (matentzn) [inline src/ontology/Makefile:454]: Never change the Makefile directly, ODK will overwrite during the next ODK update. YOu should:

Change this line:
https://github.com/ufbmi/OMRSE/blob/master/src/ontology/omrse-odk.yaml#L53 according to you suggestion here.

then run:

```
docker pull obolibrary/odkfull #too make sure you use the latest odk
sh run.sh make update_repo #run twice, the first one may error.
```

**Review by matentzn — CHANGES_REQUESTED** [review]

**Comment by hoganwr** [comment]
Bill, thank you!!!  It is an OMRSE class, so the OMRSE definition takes precedence.  If it needs editing, happy to do it, but it needs to be sourced from OMRSE.

Also, I think tying imports to specific versions of ontologies is double-edged sword.  You have finer-grained control over your dependencies (i.e., external ontologies), but then you have to be conscientious about reviewing new versions of those ontologies, and migrating to those versions (and not just flailing about wondering why they're causing issues LOL).  

How do other ODK-based ontologies handle sourcing versions of imported ontologies?

**Comment by matentzn** [comment]
@hoganwr we do not - in all OBO ontologies we work with, we try to use the latest version at all times! This makes sure that OBO ontologies are always mutually intercompatible, and issues are fixed as they arise! I would _not_ recommend using version IRIs with your imports!

**Comment by mbrochhausen** [comment]
I agree. The me previous email.

File: src/ontology/Makefile (hunk 1 above)
Comment (wdduncan) [reply src/ontology/Makefile:454]: Thanks. I forgot about the `yaml` file!

**Comment by wdduncan** [comment]
I don't have permissions to merge. But, this should be ready now.

**Review by matentzn — APPROVED** [review]

**Comment by hoganwr** [comment]
THanks for that wisdom!  I'm currently running checks, then will merge.  And yes Bill I'll give you permissions as well.

**Comment by mbrochhausen** [comment]
Hi,

I agree that the definition should be OMRSE. I also agree that importing an older version is only a second best solution. I remember that this had to do with disagreements among d-acts developers, but those issues are no longer pertinent.

Please let me know what I should change in d-acts and I’ll do it asap.

Best,
Mathias

**Pull request merged**